### PR TITLE
chore: bump cli versions

### PR DESCRIPTION
**Pull request by sobolk**
#### Description of changes

This PR:
- bumps cli versions
- adds resolution overrides in main package json to avoid conflicts with rc tags for same versions (i.e. 1.2.3-rc is resolved instead of 1.2.3-beta)
- ports changes from https://github.com/aws-amplify/amplify-cli/pull/11882 to fix e2e test failures caused by different profile initialization prompts.

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] Any CDK or CloudFormation parameter changes are called out explicitly

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: client-test-apps/js/api-model-relationship-app/src/__tests__/utils/amplifyCLI.ts [hunk 1]
```diff
@@ -145,8 +145,6 @@ const amplifyConfigure = (settings: AmplifyConfiguration): Promise<void> => {
   singleSelect(chain, s.region, amplifyRegions);
 
   return chain
-    .wait('user name:')
-    .sendCarriageReturn()
```
Comment (sobolk) [inline client-test-apps/js/api-model-relationship-app/src/__tests__/utils/amplifyCLI.ts:149]: Required by https://github.com/aws-amplify/amplify-cli/pull/11882

File: package.json [hunk 2]
```diff
@@ -139,6 +139,14 @@
...
-    "amplify-cli-core": "^4.0.0-beta.4"
+    "**/amplify-cli-core": "4.0.0-beta.5",
+    "**/@aws-amplify/amplify-environment-parameters": "1.3.4-beta.0",
+    "**/amplify-category-function": "4.3.2-beta.0",
+    "**/amplify-cli-logger": "1.2.4-beta.0",
+    "**/amplify-cli-shared-interfaces": "1.1.1-beta.3",
+    "**/amplify-function-plugin-interface": "1.9.6-beta.3",
+    "**/amplify-headless-interface": "1.17.1-beta.0",
+    "**/amplify-storage-simulator": "1.7.3-beta.0",
+    "**/amplify-util-headless-input": "1.9.10-beta.0"
```
Comment (sobolk) [inline package.json:150]: this locks packages that were incorrectly resolving to wrong tag.

File: packages/amplify-e2e-core/src/configure/index.ts [hunk 3]
```diff
@@ -42,44 +42,65 @@ const MANDATORY_PARAMS = ['accessKeyId', 'secretAccessKey', 'region'];
...
+    .wait('accessKeyId')
+    .pauseRecording()
+    .sendLine(s.accessKeyId)
+    .wait('secretAccessKey')
+    .sendLine(s.secretAccessKey)
+    .resumeRecording()
+    .wait('Profile Name:')
+    .sendLine(s.profileName)
+    .wait('Successfully set up the new user.')
+    .runAsync();
```
Comment (sobolk) [inline packages/amplify-e2e-core/src/configure/index.ts:69]: Required by https://github.com/aws-amplify/amplify-cli/pull/11882

File: packages/amplify-migration-tests/src/configure_tests.ts [hunk 4]
```diff
@@ -17,11 +18,23 @@ async function setupAmplify(version: string = 'latest') {
...
+        profileName: 'amplify-integ-test-user',
+      });
+    } else {
+      // version is before 10.7 so use the previous config function
+      await amplifyConfigureBeforeOrAtV10_7({
+        accessKeyId: AWS_ACCESS_KEY_ID,
+        secretAccessKey: AWS_SECRET_ACCESS_KEY,
+        profileName: 'amplify-integ-test-user',
+      });
+    }
```
Comment (sobolk) [inline packages/amplify-migration-tests/src/configure_tests.ts:37]: Required by https://github.com/aws-amplify/amplify-cli/pull/11882

File: package.json (hunk 2 above)
Comment (phani-srikar) [reply package.json:150]: What's the `**/` prefix in package names?

**Review by phani-srikar — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (sobolk) [reply package.json:150]: @phani-srikar it means apply to all dependency graph legs (fwiw previous definition seems to be no-op).
See https://classic.yarnpkg.com/lang/en/docs/selective-version-resolutions/#toc-how-to-use-it .

Comment (sobolk) [reply package.json:150]: see also cli definitions for comparison 
https://github.com/aws-amplify/amplify-cli/blob/259f75c681bee9b26115eec2cc5b04dca7202c0a/package.json#L142-L151

**Review by alharris-at — APPROVED** [review]

**Pull request merged**